AdyenEncryption/CardEncryptor.swift

```diff
@@ -37,161 +21,181 @@ public enum CardEncryptor {
     ///  maybe because the card public key is an invalid one, or for any other reason.
     /// - Throws:  `CardEncryptor.Error.invalidEncryptionArguments` when trying to encrypt a card with  card number, securityCode,
     /// expiryMonth, and expiryYear, all of them are nil.
-    public static func encryptedCard(for card: Card, publicKey: String) throws -> EncryptedCard {
-        guard !card.isEmpty else {
-            throw CardEncryptor.Error.invalidEncryptionArguments
+    public static func encrypt(card: UnencryptedCard, with publicKey: String) throws -> EncryptedCard {
+        guard
+            let number = card.number,
+            let securityCode = card.securityCode,
+            let expiryMonth = card.expiryMonth,
+            let expiryYear = card.expiryYear
+        else { throw Error.invalidCard }
+
+        let payloadBuilder = CardEncryptedDataBuilder(publicKey: publicKey)
```

Comment:
`EncryptedCardDataBuilder` makes more much more sense to me than `CardEncryptedDataBuilder`.

also if its a builder then it should be like this:

```
EncryptedCardDataBuilder builder = EncryptedCardDataBuilder(publicKey: publicKey)
EncryptedCardData data = try builder.add(number: number).add(securityCode: securityCode).add(expiryMonth: expiryMonth).build()

return EncryptedCard(number: try data.number,
                              securityCode: try data.securityCode,
                              expiryMonth: try data.expiryMonth,
                              expiryYear: try data.expiryYear))
```

I mean if you're going to follow the builder design pattern, now you're just using the name without the actual pattern.

this way all functions in `CardEncryptor ` will return the same type `EncryptedCardData`